crates/biome_plugin_loader/src/lib.rs

```diff
@@ -105,6 +91,37 @@ impl BiomePlugin {
     }
 }
 
+/// Normalizes the given `path` without requiring filesystem access.
+fn normalize_path(path: &Utf8Path) -> Utf8PathBuf {
```

Comment:
The meaning of the term "normalize" varies greatly by context (normalize a path VS normalise a URL). We should document precisely what this normalisation does to the paths.